packages/tpetra/core/src/Tpetra_DirectoryImpl_def.hpp

```diff
@@ -431,10 +431,28 @@ namespace Tpetra {
 
       RCP<const Teuchos::Comm<int> > comm = map.getComm ();
       const int numProcs = comm->getSize ();
-      const global_size_t nOverP = map.getGlobalNumElements () / numProcs;
       const LO LINVALID = Teuchos::OrdinalTraits<LO>::invalid();
       LookupStatus res = AllIDsPresent;
 
+      // Find the first initialized GID for search below
+      auto it = std::find_if(
+        allMinGIDs_.begin(),
+        allMinGIDs_.end()-1,
+        [](GO const gid) { return gid != std::numeric_limits<GO>::max(); }
+      );
+
+      if (it == allMinGIDs_.end()-1) {
+        // No entries have been assigned on this process
```

Comment:
Not just on this process -- on any process, right?  This should be what happens when you have an empty Map that you created with the contiguous nonuniform constructor.

More importantly, we may want to check this with an actual empty Map, created with Map's default constructor.  It sets `minMyGID_` to invalid, not the max.